### PR TITLE
Remove redundant cookies and session Id

### PR DESCRIPTION
**Pull request by magska**

### CODE REVIEW

File: autotracking/track.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ module.exports.getVisitorId = function() {
 };
 
 module.exports.getSessionId = function() {
-    return activity.getSessionId();
+    return undefined;
```
Comment (rob-mccann) [inline autotracking/track.js]: just `return` here is fine

Comment (rob-mccann) [inline autotracking/track.js]: we should probably console.log here too

File: integrationtest/test/testObjectTest.js [hunk 2]
```diff
@@ -48,9 +60,16 @@ module.exports = {
...
 
-		.deleteCookie('_DataTrackerSession')
+		.deleteCookie('_DataTrackerEnv')
 		.refresh()
 		.waitForElementVisible('body', 1000)
+
+		// Check that environment and session cookies are deleted
+
+		.getCookies(function callback(result) {
+			console.log(result.value);
```
Comment (rob-mccann) [inline integrationtest/test/testObjectTest.js]: debug code

File: integrationtest/test/testPageload.js [hunk 3]
```diff
@@ -15,29 +15,23 @@ module.exports = {
...
+			this.assert.equal(result.value.length, 2);
 
 			var count = 0;
 
 			for (var j = 0; j < result.value.length; j++) {
-				if (result.value[j].name === '_DataTrackerSession') {
-					chai.assert.match(result.value[j].value, uuidv4pattern);
-					count += 1;
-				} else if (result.value[j].name === '_DataTrackerEnv') {
+				if (result.value[j].name === '_DataTrackerEnv') {
```
Comment (rob-mccann) [inline integrationtest/test/testPageload.js]: should use `switch`.

Is there a ticket for removing these 'if' statments from the tests?

File: lib/activity.js [hunk 4]
```diff
@@ -348,7 +338,7 @@ Activity.prototype.getPageViewId = function() {
  * @returns {string} sessionId
  */
 Activity.prototype.getSessionId = function() {
-    return this.sessionId;
+    return undefined;
```
Comment (rob-mccann) [inline lib/activity.js]: as above

File: lib/user.js [hunk 5]
```diff
@@ -44,34 +41,28 @@ User.prototype.getUserId = function(callback) {
...
     var cookieId = {
-        sessionId: this.getUserIdFromCookie(this.sessionKey),
-        environmentId: this.getUserIdFromCookie(this.envKey),
-        userId: this.activity.opts.userId
+        environmentId: this.getIdsFromCookie(this.envKey)
     };
-
-    var visitorTemp = this.getUserIdFromCookie(this.visitorKey);
+    if (cookieId.environmentId === 'undefined') {
+        cookieId.environmentId = undefined;
```
Comment (rob-mccann) [inline lib/user.js]: use `delete cookieId.environmentId`

File: lib/user.js [hunk 6]
```diff
@@ -44,34 +41,28 @@ User.prototype.getUserId = function(callback) {
...
 	// Get IDs from cookies
     var cookieId = {
-        sessionId: this.getUserIdFromCookie(this.sessionKey),
-        environmentId: this.getUserIdFromCookie(this.envKey),
-        userId: this.activity.opts.userId
+        environmentId: this.getIdsFromCookie(this.envKey)
     };
-
-    var visitorTemp = this.getUserIdFromCookie(this.visitorKey);
+    if (cookieId.environmentId === 'undefined') {
```
Comment (rob-mccann) [inline lib/user.js]: where is this string set?

File: lib/user.js [hunk 7]
```diff
@@ -79,12 +70,10 @@ User.prototype.getUserId = function(callback) {
 
 			self.transferUserData(data);
```
Comment (rob-mccann) [inline lib/user.js]: should use .bind rather than self but perhaps save that for a new PR

File: lib/user.js [hunk 8]
```diff
@@ -161,37 +149,38 @@ User.prototype.getUserIdFromService = function(id, callback) {
...
- * A function that sets the anonymous user ID in a cookie.
+ * A function that sets environment and visitor ID in cookies.
  */
-User.prototype.setUserIdInCookie = function(temporary) {
+User.prototype.setIdsInCookie = function(temporary) {
 
     var now = new Date();
     var time = now.getTime();
-    now.setTime(time + 1000000 * 36000);
+    now.setTime(time + 15 * 60000);
```
Comment (rob-mccann) [inline lib/user.js]: magic number, please assign to variable or comment

File: lib/user.js [hunk 9]
```diff
@@ -161,37 +149,38 @@ User.prototype.getUserIdFromService = function(id, callback) {
...
 
     if (!temporary) {
         document.cookie = this.envKey + '=' + this.idObj.envId + ';expires=' + now.toGMTString();
     } else {
         document.cookie = this.envKey + '=' + this.idObj.envId;
     }
+    now.setTime(time + 1000000 * 36000);
```
Comment (rob-mccann) [inline lib/user.js]: as above

File: lib/user.js [hunk 10]
```diff
@@ -161,37 +149,38 @@ User.prototype.getUserIdFromService = function(id, callback) {
...
-User.prototype.setUserIdInCookie = function(temporary) {
+User.prototype.setIdsInCookie = function(temporary) {
 
     var now = new Date();
     var time = now.getTime();
-    now.setTime(time + 1000000 * 36000);
+    now.setTime(time + 15 * 60000);
 
     if (!temporary) {
         document.cookie = this.envKey + '=' + this.idObj.envId + ';expires=' + now.toGMTString();
```
Comment (rob-mccann) [inline lib/user.js]: `toGMTString` is depcrecated, use toUTCString

File: lib/user.js [hunk 11]
```diff
@@ -161,37 +149,38 @@ User.prototype.getUserIdFromService = function(id, callback) {
...
  */
-User.prototype.setSessionIdInCookie = function() {
-	var now = new Date();
-    var time = now.getTime();
-    now.setTime(time + 15 * 60000);
-	document.cookie = this.sessionKey + '=' + this.idObj.sessionId + ';expires=' + now.toGMTString();
-};
+ User.prototype.cleanUpCookies = function() {
+     if (this.getIdsFromCookie('_DataTrackerSession')) {
+         document.cookie = '_DataTrackerSession=undefined;expires=Thu, 01 Jan 1970 00:00:01 GMT';
```
Comment (rob-mccann) [inline lib/user.js]: should set to empty string rather than the string 'undefined'

Comment (rob-mccann) [inline lib/user.js]: (as you do below ;) )

File: lib/user.js (hunk 6 above)
Comment (magska) [inline lib/user.js]: Not quite sure, this is some kind of edge case that appears every once in a while. It might come from CIS. I have not found a cause for this in the SDK itself yet.

File: lib/user.js (hunk 7 above)
Comment (magska) [inline lib/user.js]: Self is more performant, and neither SPT or the internet seems to agree upon which to use. So I will stick to self for now.

File: lib/user.js (hunk 6 above)
Comment (rob-mccann) [inline lib/user.js]: see below - I think you set it to undefined when you clear it instead of empty string

File: lib/user.js (hunk 7 above)
Comment (rob-mccann) [inline lib/user.js]: self is more performant if you're doing it hundreds of thousands of calls but `bind` should be the default as it's the native way that the ecmascript standards have agreed on.

File: autotracking/track.js [hunk 12]
```diff
@@ -44,7 +44,8 @@ module.exports.getVisitorId = function() {
 };
 
 module.exports.getSessionId = function() {
-    return activity.getSessionId();
+    console.log('Deprecated: getSessionId');
```
Comment (rob-mccann) [inline autotracking/track.js]: indentation

File: lib/user.js (hunk 7 above)
Comment (magska) [inline lib/user.js]: Allright, I will create a ticket for it :) There is a lot of places to fix this in the SDK.

File: lib/activity.js [hunk 13]
```diff
@@ -377,7 +368,7 @@ Activity.prototype.refreshUserIds = function(userId) {
     } else {
         this.loggedIn = true;
     }
-    this.user.setUserIdInCookie(false);
+    this.user.setIdsInCookie(false);
```
Comment (rob-mccann) [inline lib/activity.js]: I don't see any arguments in the function definition above?

File: lib/user.js [hunk 14]
```diff
@@ -44,34 +41,28 @@ User.prototype.getUserId = function(callback) {
...
+
+    cookieId.userId = this.activity.userId;
 
 	this.transferUserData(cookieId);
     var self = this;
 
-	// If no session ID is found, do request to CIS
-	if (typeof this.idObj.sessionId === 'undefined') {
+	// If no environment ID is found, do request to CIS
+	if (typeof this.idObj.envId === 'undefined') {
```
Comment (rob-mccann) [inline lib/user.js]: as above, I think we should be doing a falsey check here checking the string undefined is going to cause issues in the future

File: lib/user.js [hunk 15]
```diff
@@ -161,37 +149,35 @@ User.prototype.getUserIdFromService = function(id, callback) {
...
+    document.cookie = this.envKey + '=' + this.idObj.envId + ';expires=' + now.toUTCString();
+    // now sould be a year from now.
+    now.setTime(time + 365 * 24 * 60 * 60 * 1000);
+    document.cookie = this.visitorKey + '=' + this.idObj.visitorId + ';expires=' + now.toUTCString();
 };
 
-/*
- * SessionId should last for 15 minutes in cookie.
+/**
+ * Function that removes redundatn cookies
```
Comment (rob-mccann) [inline lib/user.js]: typo

File: lib/user.js [hunk 16]
```diff
@@ -161,37 +149,35 @@ User.prototype.getUserIdFromService = function(id, callback) {
...
  */
-User.prototype.setSessionIdInCookie = function() {
-	var now = new Date();
-    var time = now.getTime();
-    now.setTime(time + 15 * 60000);
-	document.cookie = this.sessionKey + '=' + this.idObj.sessionId + ';expires=' + now.toGMTString();
-};
+ User.prototype.cleanUpCookies = function() {
+     if (this.getIdsFromCookie('_DataTrackerSession')) {
```
Comment (rob-mccann) [inline lib/user.js]: do you need this check? can't we just set it?

Comment (magska) [inline lib/user.js]: It is a temporary function to delete cookies that are no longer used due to an issue with finn.no and their loadbalancer not handeling more than some amounts of cookies. I have added to the description that is should be removed at some point. (I know, this is not an ideal solution).

File: lib/activity.js [hunk 17]
```diff
@@ -199,12 +195,6 @@ Activity.prototype.addUserId = function(object) {
         object.actor['spt:environmentId'] = this.envId;
     }
 
-    if (this.sessionId !== null && String(this.sessionId).indexOf('urn') === -1) {
-        object.actor['spt:sessionId'] = 'urn:schibsted.com:session:' + this.sessionId;
-    } else {
-        object.actor['spt:sessionId'] = this.sessionId;
-    }
-
     if (this.userId && String(this.userId).indexOf('undefined') === -1) {
```
Comment (rob-mccann) [inline lib/activity.js]: string undefined check here

Comment (magska) [inline lib/activity.js]: Could you rephrase that? As this value is provided by clients, I am doing a falsy check and checking if any part of the string contains the text "undefined".

Comment (rob-mccann) [inline lib/activity.js]: my bad, :+1:

Comment (rob-mccann) [inline lib/activity.js]: needs a comment anywhere we do anything with the string undefined in this context

**Comment by magska** [comment]
@rob-mccann anything else before I put this in production?

**Comment by rob-mccann** [comment]
as long as it's backwards compatible, :+1:
